index.d.ts

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import Store from '@orbit/store';
+import { TransformOrOperations, QueryBuilder, Source } from '@orbit/data';
+
+
+export interface DataProviderProps {
+  dataStore: Store;
+}
+
+export interface RecordsToProps {
+  [key: string]: (q: QueryBuilder) => any;
+}
+
+export interface WithData {
+  dataStore: Store;
+  sources: { [sourceName: string]: Source };
```

Comment:
please remove from this PR (it's independent from type support) and open separate PR for adding sources convenience prop

Reply:
I'll open a different PR with just the types :)